Expose more fields in OperatingSystemStatus

toanju opened the pull request:
Expose more fields in OperatingSystemStatus

Exposes the following additional fields

* GardenLinuxCommitID
* GardenLinuxFeatures
* VariantID

to share more details on a Garden Linux based installation

fwiesel commented:
I see the need in it, and it is probably just bike-shedding, but this is tied to the gardenlinux build process, and doesn't cover all the inputs (i.e. GARDENLINUX_CNAME).

How about setting and using some [standard fields](https://www.freedesktop.org/software/systemd/man/latest/os-release.html) instead?
I'd be a fan of setting `BUILD_ID`  to `${{github.workflow}}-${{github.run_number}}` or `${{github.run_id}}`.

Or would `VARIANT_ID` and `GARDENLINUX_COMMIT_ID` suffice to uniquely identify the build?

fwiesel reviewed (commented):
I'll approve it, in case you want to move forward regardless of the alternatives proposed.
Worst-case we migrate..

toanju commented:
we were discussing if `BUILD_ID` actually suites here, however [BUILD_ID](https://www.freedesktop.org/software/systemd/man/latest/os-release.html#BUILD_ID=) does not really match for Garden Linux since we do not have a particular underlying build that resides the same. Will update to `commitID`.

notandy commented on api/v1/hypervisor_types.go:
```diff
@@ -123,6 +123,9 @@ type OperatingSystemStatus struct {
 	Version string `json:"version,omitempty"`
 
+	// Represents the Operating System build commit SHA sum.
+	CommitID string `json:"commit_id,omitempty"`
```
```suggestion
	CommitID string `json:"commitID,omitempty"`
```

toanju replied:
addressed, ty

toanju committed:
Expose more fields in OperatingSystemStatus

Exposes the following additional fields

* GardenLinuxCommitID
* GardenLinuxFeatures
* VariantID

to share more details on a Garden Linux based installation

anokfireball commented on api/v1/hypervisor_types.go:
```diff
@@ -152,6 +155,12 @@ type OperatingSystemStatus struct {
+	GardenLinuxCommitID string `json:"gardenLinuxCommitID,omitempty"`
+
+	// Represents the Garden Linux Feature Set
+	GardenLinuxFeatures string `json:"gardenLinuxFeatures,omitempty"`
```
```suggestion
	GardenLinuxFeatures []string `json:"gardenLinuxFeatures,omitempty"`
```

anokfireball reviewed (approved):
Maybe it is worth it to "properly" expose the feature list? Don't feel forced to follow through with this suggestion, though!

toanju commented on api/v1/hypervisor_types.go:
```diff
@@ -179,6 +155,12 @@ type OperatingSystemStatus struct {
+	GardenLinuxCommitID string `json:"gardenLinuxCommitID,omitempty"`
+
+	// Represents the Garden Linux Feature Set
+	GardenLinuxFeatures *[]string `json:"gardenLinuxFeatures,omitempty"`
```
fixed
